### DISCUSSION TITLE
External Library:ro

### DISCUSSION THREAD

**Discussion by mxm199** [immich-app/immich#14117] [General]
If the external library is mounted in **read-only mode**, then some functions of working with the library, such as working with tags, are unavailable, since of course an **xmp** file cannot be created in the library. 
Is it possible for xmp files to be created in another mounted folder in read-write mode?

**Comment by zackpollard** [comment]
I'm not sure we've discussed this, but it's an interesting idea. We have more changes queued to be made to external libraries in future, so I will try to keep this in mind as a potential idea.

**Comment by magicalpig** [comment]
I discovered the "exif" database table and was expecting to see my manually added metadata ("Description") populated into this table. Instead the system attempted to write an XMP file to the read-only external library. This had me wondering what the purpose of the "exif" table is. Is it for the metadata scraped from the asset when it was first discovered?

**Reply by bo0tzz** [reply]
It is for metadata scraped from the asset+sidecar. That happens on first ingest of course, but also for example after a metadata edit writes to the XMP sidecar.

**Reply by magicalpig** [reply]
Thank you. If my xmp write had been successful (it failed because the asset resided in a read-only external library) I suppose the server would have updated the exif database table as its next action.

**Comment by kvalev** [comment]
It would be nice if the sidecar folder was configurable (à la photoprism), as discussed here https://github.com/immich-app/immich/discussions/5530

**Comment by skatsubo** [comment]
As suggested here https://github.com/immich-app/immich/discussions/18502#discussioncomment-13238801 OverlayFS can be used to allow writing XMP sidecars for RO library, though it requires some filesystem crafting.

**Reply by kvalev** [reply]
Its not really that difficult, its just a few lines in the docker compose file. Here is an example:
https://github.com/immich-app/immich/discussions/5530#discussioncomment-12001052

**Reply by SinTan1729** [reply]
But it's not perfect. Rescans won't remove deleted files unless Immich is restarted.
